tensorboard/plugins/hparams/summary.py

```diff
@@ -12,68 +12,136 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
-"""Utilities for creating summaries for experiments and sessions
+"""Summary creation methods for the HParams plugin.
+
+Typical usage for exporting summaries in a hyperparameters-tuning experiment:
+1. Create the experiment (once) by calling experiment_pb() and exporting
+   the resulting summary into a top-level (empty) run.
+2. In each training session in the experiment, call session_start_pb() before
+   the session starts, exporting the resulting summary into a uniquely named
+   run for the session, say <session_name>.
+3. Train the model in the session, exporting each metric as a scalar summary
+   in runs of the form <session_name><suffix>, where suffix depends on the
+   metric, and may be empty. The name of such a metric is a (group, tag) pair
+   given by (<suffix>, tag) where tag is the tag of the scalar summary.
+   When calling experiment_pb in step 1, you'll need to pass all the metric
+   names used in the experiemnt.
```

Comment:
One comment: we don't any notion of metrics in the experiment itself: e.g. users can provide arbitrary metric timestamp:value pairs at any point in time; we don't enforce any notion of metric names that have to be defined ahead of time.
We could easily construct an experiment_pb based on existing runs (e.g. fetch the union of all existing metric names); but only for the first run- we won't have any metrics yet. This shouldn't be a problem, right?